DataManagementSystem/Agent/RequestOperations/MoveReplica.py

```diff
@@ -7,201 +7,324 @@
     self.ci = ConsistencyInspector()
 
-  def __call__( self ):
+  def __call__(self):
     """ call me maybe """
+
+    # Check whether the ES flag is enabled so we can send the data accordingly.
+    if self.rmsMonitoring:
+      self.rmsMonitoringReporter = MonitoringReporter(monitoringType="RMSMonitoring")
+    else:
+      # # own gMonitor stuff for files
+      gMonitor.registerActivity("ReplicateAndRegisterAtt", "Replicate and register attempted",
+                                "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM)
+      gMonitor.registerActivity("ReplicateOK", "Replications successful",
+                                "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM)
+      gMonitor.registerActivity("ReplicateFail", "Replications failed",
+                                "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM)
+      gMonitor.registerActivity("RegisterOK", "Registrations successful",
+                                "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM)
+      gMonitor.registerActivity("RegisterFail", "Registrations failed",
+                                "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM)
+      gMonitor.registerActivity("RemoveReplicaAtt", "Replica removals attempted",
+                                "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM)
+      gMonitor.registerActivity("RemoveReplicaOK", "Successful replica removals",
+                                "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM)
+      gMonitor.registerActivity("RemoveReplicaFail", "Failed replica removals",
+                                "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM)
+
     # # check replicas first
     res = self.__checkReplicas()
     if not res["OK"]:
-      self.log.error( 'Failed to check replicas', res["Message"] )
+      self.log.error('Failed to check replicas', res["Message"])
 
     sourceSE = self.operation.SourceSE if self.operation.SourceSE else None
     if sourceSE:
       # # check source se for read
-      bannedSource = self.checkSEsRSS( sourceSE, 'ReadAccess' )
+      bannedSource = self.checkSEsRSS(sourceSE, 'ReadAccess')
       if not bannedSource["OK"]:
-        gMonitor.addMark( "ReplicateAndRegisterAtt", len( self.operation ) )
-        gMonitor.addMark( "ReplicateFail", len( self.operation ) )
+        if self.rmsMonitoring:
+          for opFile in self.operation:
+            for status in ["FileAttempted", "FileFailed"]:
+              self.rmsMonitoringReporter.addRecord({
+                  "timestamp": time.mktime(datetime.datetime.utcnow().timetuple()),
+                  "host": socket.getfqdn(),
+                  "objectType": "File",
+                  "operationType": self.operation.Type,
+                  "objectID": opFile.FileID,
+                  "parentID": self.operation.OperationID,
+                  "status": status,
+                  "nbObject": 1
```

Comment:
I think the idea for File was to not use the `objectID`, thus allowing you to send only one report with `nbObject` being the total number of files. @zmathe can you confirm ?

Reply:
yes, right. We want to send one report with total number of files